daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -852,56 +856,63 @@ private[lf] object Speedy {
   private[speedy] sealed trait Kont {
 
     /** Execute the continuation. */
-    def execute(v: SValue, machine: Machine): Unit
+    def execute(v: SValue): Unit
   }
 
   /** Final continuation; machine has computed final value */
-  private[speedy] final case object KFinished extends Kont {
-    def execute(v: SValue, machine: Machine) = {
+  private[speedy] final case class KFinished(machine: Machine) extends Kont {
```

Comment:
Having access to the `machine` in this continuation seems to cause a lot of trouble in the rest of the PR. Can't we do the instrumentation stuff in the exception handler for what we throw here instead? That seems to significantly simplify things.

Reply:
maybe, I'll have a look
(But I don't really see it as that much trouble.)

Reply:
Moved the instrumentation print into `run()`, avoiding need for `KFinished` to have access to `machine`.
I wouldn't say this significantly simplifies things, but is does reduce the diff noise in this PR.